### PR TITLE
Capture Ctrl+C in cli_wallet when not in daemon mode #1193

### PR DESCRIPTION
**Pull request by cogutvalera**
PR for #1193

### CODE REVIEW

File: programs/cli_wallet/main.cpp [hunk 1]
```diff
@@ -279,19 +279,20 @@ int main( int argc, char** argv )
       {
          wallet_cli->register_api( wapi );
          wallet_cli->start();
-         wallet_cli->wait();
+         //wallet_cli->wait();
```
Comment (abitmore) [inline programs/cli_wallet/main.cpp]: Please remove it.

**Comment by abitmore** [comment]
After this change, what will happen when pressed ctrl+c in cli_wallet?

**Comment by cogutvalera** [comment]
valera@valera-GS70-2OD:~/Projects/Blockchain/Graphene/BitShares/bitshares-core$ ./programs/cli_wallet/cli_wallet --server-rpc-endpoint=wss://node.bitshares.eu
Logging RPC to file: logs/rpc/rpc.log
3334817ms th_a       main.cpp:143                  main                 ] key_to_wif( committee_private_key ): 5KCBDTcyDqzsqehcb52tW5nU6pXife6V2rX9Yf7c3saYSzbDZ5W
3334817ms th_a       main.cpp:147                  main                 ] nathan_pub_key: BTS6MRyAjQq8ud7hVNYcfnVPJqcVpscN5So8BhtHuGYqET5GDW5CV
3334818ms th_a       main.cpp:148                  main                 ] key_to_wif( nathan_private_key ): 5KQwrPbwdL6PhXujxW37FSSQZ1JiwsST4cqQzDeyXtP79zkvFD3
3334819ms th_a       main.cpp:195                  main                 ] wdata.ws_server: wss://node.bitshares.eu
3335013ms th_a       main.cpp:200                  main                 ] wdata.ws_user:  wdata.ws_password:
locked >>>

3336605ms th_a       wallet.cpp:805                save_wallet_file     ] saving wallet to file wallet.json
9 canceled_exception: Canceled
cancellation reason: [none given]
    {"reason":"[none given]"}
    th_a  thread_d.hpp:473 start_next_fiber
pure virtual method called
terminate called without an active exception
Aborted (core dumped)

**Comment by cogutvalera** [comment]
now in cli_wallet CTRL+C has the same behaviour as QUIT command in cli_wallet

**Comment by cogutvalera** [comment]
what will happen after QUIT command in cli_wallet:

```
valera@valera-GS70-2OD:~/Projects/Blockchain/Graphene/BitShares/bitshares-core$ ./programs/cli_wallet/cli_wallet --server-rpc-endpoint=wss://node.bitshares.eu
Logging RPC to file: logs/rpc/rpc.log
3456791ms th_a       main.cpp:143                  main                 ] key_to_wif( committee_private_key ): 5KCBDTcyDqzsqehcb52tW5nU6pXife6V2rX9Yf7c3saYSzbDZ5W
3456792ms th_a       main.cpp:147                  main                 ] nathan_pub_key: BTS6MRyAjQq8ud7hVNYcfnVPJqcVpscN5So8BhtHuGYqET5GDW5CV
3456792ms th_a       main.cpp:148                  main                 ] key_to_wif( nathan_private_key ): 5KQwrPbwdL6PhXujxW37FSSQZ1JiwsST4cqQzDeyXtP79zkvFD3
3456792ms th_a       main.cpp:195                  main                 ] wdata.ws_server: wss://node.bitshares.eu
3457034ms th_a       main.cpp:200                  main                 ] wdata.ws_user:  wdata.ws_password:
locked >>> quit
quit
3458738ms th_a       wallet.cpp:786                quit                 ] Quitting Cli Wallet ...
null
3458738ms th_a       wallet.cpp:805                save_wallet_file     ] saving wallet to file wallet.json
9 canceled_exception: Canceled
cancellation reason: [none given]
    {"reason":"[none given]"}
    th_a  thread_d.hpp:473 start_next_fiber
pure virtual method called
terminate called without an active exception
Aborted (core dumped)
```

**Comment by abitmore** [comment]
Perhaps revert the FC `canceled_exception` change as well?

**Comment by cogutvalera** [comment]
yes absolutely right :+1:  There is already done PR in FC :-) https://github.com/bitshares/bitshares-fc/pull/68

**Comment by abitmore** [comment]
Thanks. Please rebase and bump fc version to latest master.

**Comment by abitmore** [comment]
Oh, wait. It's not correct. Raising SIGINT will terminate cli_test early.

**Comment by cogutvalera** [comment]
so is it better to throw exception as was before ?

**Comment by cogutvalera** [comment]
reverting my changes to throw exception in cli_wallet quit command and in FC too as was before in order not to terminate cli_test early

**Comment by cogutvalera** [comment]
sorry forgot about cli_test, thank you very much for remembering it to me :+1:

**Comment by cogutvalera** [comment]
https://github.com/bitshares/bitshares-fc/pull/70

**Comment by abitmore** [comment]
Please revert the FC bump.

**Comment by abitmore** [comment]
Thank YOU indeed.

**Pull request merged**